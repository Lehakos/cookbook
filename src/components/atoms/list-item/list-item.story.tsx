import React from 'react';
import { storiesOf } from '@storybook/react';

import { ListItem } from './list-item';

const stories = storiesOf('Atoms|ListItem', module);

stories.add('Default', () => <ListItem>Hello</ListItem>);
