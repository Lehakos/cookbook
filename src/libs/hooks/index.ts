export { useControlledProp } from './use-controlled-prop';
export { useElementSize } from './use-element-size';
