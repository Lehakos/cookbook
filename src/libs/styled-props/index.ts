export { typography, TypographyProps } from './typography';
export { layout, LayoutProps } from './layout';
export { flexbox, FlexboxProps } from './flexbox';
