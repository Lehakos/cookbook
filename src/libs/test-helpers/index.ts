export { render } from './render';
export { mockElementWithBoundingRect } from './mock-element-with-bounding-rect';
