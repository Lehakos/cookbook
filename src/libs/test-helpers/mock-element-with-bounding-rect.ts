export const mockElementWithBoundingRect = ({
  width = 100,
  height = 100,
  top = 0,
  left = 0,
}) => {
  const right = left + width;
  const bottom = top + height;

  Object.defineProperties(Element.prototype, {
    offsetLeft: {
      get() {
        return left;
      },
      configurable: true,
    },
    offsetTop: {
      get() {
        return top;
      },
      configurable: true,
    },
    offsetHeight: {
      get() {
        return height;
      },
      configurable: true,
    },
    offsetWidth: {
      get() {
        return width;
      },
      configurable: true,
    },
  });

  Element.prototype.getBoundingClientRect = jest.fn(() => {
    return {
      width,
      height,
      top,
      left,
      bottom,
      right,
    };
  });

  const element = document.createElement('div');

  return element;
};
