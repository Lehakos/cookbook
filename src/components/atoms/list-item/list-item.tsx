import React from 'react';

import * as s from './list-item.styles';

interface Props {
  oneLine?: boolean;
}

export const ListItem: React.FunctionComponent<Props> = function({
  children,
  oneLine,
}) {
  return (
    <s.Wrapper>
      <s.Bullet />
      <s.Text oneLine={oneLine}>{children}</s.Text>
    </s.Wrapper>
  );
};
