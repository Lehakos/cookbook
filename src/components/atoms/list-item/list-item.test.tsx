import React from 'react';

import { render } from 'libs/test-helpers';

import { ListItem } from './list-item';

describe('<ListItem />', () => {
  it('renders properly', () => {
    const { container } = render(<ListItem />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
