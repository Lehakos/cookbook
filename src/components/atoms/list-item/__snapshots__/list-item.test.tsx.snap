// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem /> renders properly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
}

.c0:not(:last-child) {
  margin-bottom: 4px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin-right: 8px;
}

.c1::before {
  content: 'A';
  width: 0;
  visibility: hidden;
}

.c1::after {
  height: 4px;
  width: 4px;
  content: '';
  border-radius: 50%;
  background: #F47983;
}

<li
  class="c0"
>
  <span
    class="c1"
  />
  <span
    class=""
  />
</li>
`;
